### PR TITLE
[JS] onnxruntime-web

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**: This change add code for `onnxruntime-web`.

onnxruntime-web is designed to run ONNXRuntime in web browser environment. It supports 2 execution providers (aka Backend): wasm and webgl.

wasm backend: uses ONNXRuntime WebAssembly
webgl backend: uses onnxjs

build:

1. preparation (build wasm)
    1. in root folder, use `build --build_wasm` to build ORT web assembly.
    2. copy files `onnxruntime_wasm.*` from build output folder to `/js/web/lib/wasm/binding/`.

2. install NPM packages
    1. in `/js/`, run `npm ci`.
    2. in `/js/common/`, run `npm ci`.
    3. in `/js/web/`, run `npm ci`.

3. build or test in `/js/web/`
    - `npm run build`
    - `npm test`

### CODE REVIEW

File: js/node/README.md [hunk 1]
```diff
@@ -10,12 +10,6 @@ Install the latest stable version:
 npm install onnxruntime
```
Comment (hanbitmyths) [inline js/node/README.md:10]: onnxruntime-node?

File: js/web/lib/onnxjs/attribute.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    }
+  }
+
+  set(key: string, type: Attribute.DataType, value: ValueTypes): void {
+    this._attributes.set(key, [value, type]);
+  }
+  delete(key: string): void {
+    this._attributes.delete(key);
+  }
+  getFloat(key: string, defaultValue?: Attribute.DataTypeMap['float']) {
```
Comment (hanbitmyths) [inline js/web/lib/onnxjs/attribute.ts:49]: line break

File: js/web/lib/wasm/binding/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import wasmModuleFactory, {BackendWasmModule} from './onnxruntime_wasm';
```
Comment (hanbitmyths) [inline js/web/lib/wasm/binding/index.ts:1]: multi-threading support is not in this PR?

File: js/web/script/test-runner-cli-args.ts [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+Run ONNX.js tests, models, benchmarks in different environments.
+
+Usage:
+ test-runner-cli <mode> ... [options]
+
+Modes:
+ suite0                        Run all unittests, all operator tests and node model tests that described in white list
+ suite1                        Run all unittests, all operator tests and all model tests that described in white list
+ model                         Run a single model test
```
Comment (hanbitmyths) [inline js/web/script/test-runner-cli-args.ts:25]: We don't support model test directly at onnxruntime. Put a comment about how to run a model test.

File: js/web/lib/onnxjs/attribute.ts [hunk 5]
```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (fs-eire) [reply js/web/lib/onnxjs/attribute.ts:1]: yes, this is not the latest version. I picked an early version in this integration.

File: js/web/lib/wasm/binding/index.ts (hunk 3 above)
Comment (fs-eire) [reply js/web/lib/wasm/binding/index.ts:1]: no, but I am working on with it

File: js/common/README.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# ONNX Runtime JavaScript API
+
+ONNX Runtime JavaScript API is a unified API for all JavaScript usages. It's dependency of the following NPM packages:
+
+- onnxruntime-node
+- onnxruntime-web
+- onnxruntime-react-native
+
+ONNX Runtime JavaScript API is not designed for using directly. Please consider to install one of the NPM packages above according to target platform.
```
Comment (hanbitmyths) [inline js/common/README.md:9]: Actually, a user will use ONNX Runtime JavaScript API directly. It's not designed for standalone library?

File: js/web/README.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+This is the most straightforward way to use ONNX Runtime Web. The following HTML example shows how to use it:
+
+```html
+<html>
+  <head> </head>
+
+  <body>
+    <!-- Load ONNX Runtime Web -->
+    <script src="https://cdn.jsdelivr.net/npm/onnxruntime-web/dist/ort.min.js"></script>
```
Comment (hanbitmyths) [inline js/web/README.md:33]: Did you change ort.min.js to ort-web.min.js in other commit?

File: js/common/README.md (hunk 6 above)
Comment (fs-eire) [reply js/common/README.md:9]: This library, onnxruntime-common itself, is a standalone library. it does not have extra dependency for itself to work correctly

"not designed for using directly" here means users usually want to have at least one backend implementation to actually inference an onnx model.

File: js/web/README.md (hunk 7 above)
Comment (fs-eire) [reply js/web/README.md:33]: not in examples. it's in onnxruntime-web's package.json

Comment (fs-eire) [reply js/web/README.md:33]: actually, ort.min.js = ort-common.min.js + ort-web.min.js

File: js/common/README.md (hunk 6 above)
Comment (fs-eire) [reply js/common/README.md]: updated readme.

**Review by hanbitmyths — APPROVED** [review]

**Pull request merged**